src/mlpack/methods/random_forest/extra_trees_main.cpp

```diff
@@ -0,0 +1,278 @@
+        true,
+        "minimum leaf size must be greater than 0");
+    RequireParamValue<int>(
+        "maximum_depth", [](int x)
+        { return x >= 0; },
+        true,
+        "maximum depth must not be negative");
+    RequireParamValue<int>(
+        "subspace_dim", [](int x)
+        { return x >= 0; },
+        true,
+        "subspace dimensionality must be nonnegative");
+    RequireParamValue<double>(
+        "minimum_gain_split",
+        [](double x)
+        { return x >= 0.0; },
+        true,
+        "minimum gain for splitting must be nonnegative");
+    ReportIgnoredParam({{"training", false}}, "num_trees");
+    ReportIgnoredParam({{"training", false}}, "minimum_leaf_size");
+    ExtraTreesModel *etModel;
+    if (IO::HasParam("input_model"))
+        model = IO::GetParam << ExtraTreesModel > ("input_model");
+    else
+        etmodel = new ExtraTreesModel();
+    if (IO ::HasParam("training"))
+    {
+        Timer::Start("et_training");
+        //Training the model on the given input data.
+        mat data = std::move(IO::.GetParam<mat>("training"));
+        Row<size_t> labels = std::move(IO::GetParam<Row<size_t>>("labels"));
+        // Make sure the subspace dimensionality is valid.
+        RequireParamValue<int>(
+            "subspace_dim",
+            [data](int x)
+            { return (size_t)x <= data.n_rows; },
+            true, "subspace "
+                "dimensionality must not be greater than data dimensionality");
+    const size_t numTrees = (size_t)IO::GetParam<int>("num_trees");
+    const size_t minimumLeafSize = 
+    (size_t)IO::GetParam<int>("minimum_leaf_size");
+    const size_t maxDepth = (size_t)IO::GetParam<int>("maximum_depth");
+    const double minimumGainSplit = IO::GetParam<double>("minimum_gain_split");
+    const size_t randomDims = (IO::GetParam<int>("subspace_dim") == 0) ?
+        (size_t)std::sqrt(data.n_rows) :
+        (size_t)IO::GetParam<int>("subspace_dim");
+        MultipleRandomDimensionSelect mrds(randomDims);
+
+        Log::Info << "Training extra trees with " << numTrees << " trees..." 
+        << endl;
+        const size_t numClasses = max(labels) + 1;
+        //Training the model.
+    etmodel->model.Train(data, labels, numClasses, numTrees, minimumLeafSize, 
+    minimumGainSplit, maxDepth, IO::HasParam("warm_sta  rt"));
+    Timer::Stop("et_training");
+    if (IO::HasParam("test"))
+    {
+        mat testData = move(IO::GetParam<mat>("test"));
+        Timer::Start("rf_prediction");
+        //Get Predictions and probabilities.
+        ROW<size_t> predictions;
+        mat probabilities;
+        etmodel->et.Classify(testData, predictions, probabilities);
+        }
+        //Save the outputs.
+        IO::GetParam<mat>("probabilities") = std::move(probabilities);
+        IO::GetParam<Row<size_t>>("predictions") = std::move(predictions);
+    }
+    //Save the output model.
+    IO::GetParam<ExtraTreesModel *>("output_model") = etmodel;
```

Comment:
Looks like the indentation went crazy here, you can check https://github.com/mlpack/mlpack/wiki/DesignGuidelines for some more details about the coding style.